services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/VxlanFlowSpec.groovy

```diff
@@ -0,0 +1,396 @@
+
+        when: "Create a flow with protected path"
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.allocateProtectedPath = true
+        flow.maximumBandwidth = bandwidth
+        flow.ignoreBandwidth = bandwidth == 0
+        flow.source.vlanId = vlanId
+        flow.encapsulationType = FlowEncapsulationType.VXLAN
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with protected path"
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        flowPathInfo.protectedPath
+        northbound.getFlow(flow.id).flowStatusDetails
+        def flowInfoFromDb = database.getFlow(flow.id)
+        def protectedForwardCookie = flowInfoFromDb.protectedForwardPath.cookie.value
+        def protectedReverseCookie = flowInfoFromDb.protectedReversePath.cookie.value
+
+        and: "Rules for main and protected paths are created"
+        Wrappers.wait(WAIT_OFFSET) { flowHelper.verifyRulesOnProtectedFlow(flow.id) }
+
+        //TODO(andriidovhan) uncomment when pr2530 is merged
+//        and: "Validation of flow must be successful"
+//        northbound.validateFlow(flow.id).each { direction ->
+//            assert direction.discrepancies.empty
+//        }
+
+        when: "Update flow: disable protected path(allocateProtectedPath=false)"
+        def protectedFlowPath = northbound.getFlowPath(flow.id).protectedPath.forwardPath
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = false })
+
+        then: "Protected path is disabled"
+        !northbound.getFlowPath(flow.id).protectedPath
+        !northbound.getFlow(flow.id).flowStatusDetails
+
+        and: "Rules for protected path are deleted"
+        Wrappers.wait(WAIT_OFFSET) {
+            protectedFlowPath.each { sw ->
+                def rules = northbound.getSwitchRules(sw.switchId).flowEntries.findAll {
+                    !Cookie.isDefaultRule(it.cookie)
+                }
+                assert rules.every { it != protectedForwardCookie && it != protectedReverseCookie }
+            }
+        }
+
+          //TODO(andriidovhan) uncomment when pr2530 is merged
+//        and: "Validation of flow must be successful"
+//        northbound.validateFlow(flow.id).each { direction ->
+//            assert direction.discrepancies.empty
+//        }
+
+        and: "Cleanup: Delete the flow and reset costs"
+        flowHelper.deleteFlow(flow.id)
+
+        where:
+        bandwidth | vlanId
+        1000      | 3378
+        0         | 0
+    }
+
+    @Tags(HARDWARE)
+    def "System allows tagged traffic via default flow(0<->0) with 'VXLAN' encapsulation"() {
+        // we can't test (0<->20, 20<->0) because iperf is not able to establish a connection
+        given: "Noviflow switches"
+        def noviflowSwitchesIds = topology.activeSwitches.findAll { it.noviflow }*.dpId
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected.findAll {
```

Comment:
Remove `noviflowSwitchesIds`
Do
`def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected.findAll { it.noviflow }`